cub/cub/device/dispatch/dispatch_reduce.cuh

```diff
@@ -44,19 +44,18 @@
 #  pragma system_header
 #endif // no system header
 
-#include <cub/agent/agent_reduce.cuh>
 #include <cub/detail/launcher/cuda_runtime.cuh>
 #include <cub/device/dispatch/kernels/reduce.cuh>
+#include <cub/device/dispatch/kernels/segmented_reduce.cuh>
 #include <cub/device/dispatch/tuning/tuning_reduce.cuh>
 #include <cub/grid/grid_even_share.cuh>
-#include <cub/iterator/arg_index_input_iterator.cuh>
 #include <cub/thread/thread_operators.cuh>
 #include <cub/thread/thread_store.cuh>
 #include <cub/util_debug.cuh>
 #include <cub/util_device.cuh>
 #include <cub/util_temporary_storage.cuh>
 
-#include <iterator>
+// #include <iterator>
```

Comment:
I looked through the source file and you need for the iterator utilities used here:
```suggestion
#include <cub/util_type.cuh>
```

Reply:
Yes, indeed. It is needed for ``cub::detail::non_void_value_t` and ``cub::detail::value_t``. 

The source file also uses ``cub::detail::invoke_result_t`` which is declared in "cub/detail/type_traits.cuh", but this header is not included here. Should it be?

Reply:
Please do so! Always include with angle brackets though:
```c++
#include <cub/detail/type_traits.cuh>
```